src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -241,11 +252,16 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
 
                 case TableExpressionKind.Top:
                     var sortOrder = context.GetFirstSortOrderForSupportedParam();
+                    var (paramInfo, sort) = context.GetFirstSupportedSortParam();
```

Comment:
Since we are getting paramInfo, sort from context.GetFirstSupportedSortParam, the line above (sortOrder = context.GetFirstSortOrderForSupportedParam();) looks redundant. I think sort and sortOrder should be same here.